src/pages/patientView/PatientViewPage.tsx

```diff
@@ -538,18 +539,19 @@ export default class PatientViewPage extends React.Component<IPatientViewPagePro
                     </MSKTab>
                     )}
 
-                    {/*<MSKTab key={5} id="mutationalSignatures" linkText="Mutational Signature Data" hide={true}>*/}
-                        {/*<div className="clearfix">*/}
-                            {/*<FeatureTitle title="Mutational Signatures" isLoading={ patientViewPageStore.clinicalDataGroupedBySample.isPending } className="pull-left" />*/}
-                            {/*<LoadingIndicator isLoading={patientViewPageStore.mutationalSignatureData.isPending}/>*/}
-                            {/*{*/}
-                                {/*(patientViewPageStore.clinicalDataGroupedBySample.isComplete && patientViewPageStore.mutationalSignatureData.isComplete) && (*/}
-                                    {/*<ClinicalInformationMutationalSignatureTable data={patientViewPageStore.mutationalSignatureData.result} showTitleBar={true}/>*/}
-                                {/*)*/}
-                            {/*}*/}
-                        {/*</div>*/}
-
-                    {/*</MSKTab>*/}
+                    {(patientViewPageStore.mutationalSignatureMolecularProfileId.isComplete && patientViewPageStore.mutationalSignatureMolecularProfileId.isComplete && patientViewPageStore.mutationalSignatureData.isComplete) && (
+                        <MSKTab key={7} id="mutationalSignatures" linkText="Mutational Signature Data" hide={!patientViewPageStore.mutationalSignatureMolecularProfileId.result || patientViewPageStore.mutationalSignatureData.result.length === 0}>
+                            <div className="clearfix">
+                                <FeatureTitle title="Mutational Signatures" isLoading={ patientViewPageStore.clinicalDataGroupedBySample.isPending } className="pull-left" />
+                                <LoadingIndicator isLoading={patientViewPageStore.mutationalSignatureData.isPending}/>
+                                {
+                                    (patientViewPageStore.clinicalDataGroupedBySample.isComplete && patientViewPageStore.mutationalSignatureData.isComplete && patientViewPageStore.mutationalSignatureData.result) && (
```

Comment:
if there is really a case where result is falsy (null or undefined), then we should handle it here in code.  Otherwise, it's safe to assume that it's filled and you can use ! to tell typescript that it exists.

Reply:
@alisman Thanks! I think I can remove it since if the result is falsy, this tab will be hidden. I will update this later.